inference-engine/src/mkldnn_plugin/nodes/mkldnn_adaptive_pooling.cpp

```diff
@@ -0,0 +1,363 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_adaptive_pooling.h"
+#include "ie_parallel.hpp"
+#include <cpu/x64/cpu_isa_traits.hpp>
+#include <math.h>
+#include <mkldnn.hpp>
+#include <mkldnn_extension_utils.h>
+#include <mkldnn_selective_build.h>
+#include <mkldnn_types.h>
+#include <ngraph/opsets/opset8.hpp>
+#include <string>
+#include <utils/bfloat16.hpp>
+#include <utils/general_utils.h>
+#include <vector>
+
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+using namespace mkldnn;
+using namespace mkldnn::impl::cpu;
+using namespace mkldnn::impl::cpu::x64;
+
+
+bool MKLDNNAdaPoolingNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (one_of(op->get_type_info(), ngraph::op::v8::AdaptiveAvgPool::type_info)) {
+            auto adaPool = std::dynamic_pointer_cast<ngraph::opset8::AdaptiveAvgPool>(op);
+            if (!adaPool) {
+                errorMessage = "Only opset8 AdaptiveAvgPooling operation is supported";
+                return false;
+            }
+        } else if (one_of(op->get_type_info(), ngraph::op::v8::AdaptiveMaxPool::type_info)) {
+            auto adaPool = std::dynamic_pointer_cast<ngraph::opset8::AdaptiveMaxPool>(op);
+            if (!adaPool) {
+                errorMessage = "Only opset8 AdaptiveMaxPooling operation is supported";
+                return false;
+            }
+        } else {
+            errorMessage = "Unsupported Adaptive pooling mode";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+MKLDNNAdaPoolingNode::MKLDNNAdaPoolingNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng,
+                                           MKLDNNWeightsSharing::Ptr &cache) : MKLDNNNode(op, eng, cache) {
+    auto maxPoolOp = ngraph::as_type_ptr<ngraph::op::v8::AdaptiveMaxPool>(op);
```

Comment:
No need to cast to check operation type. Use get_type_info() instead.

Reply:
Done